BeeSwift/Components/GoalImageView.swift

```diff
@@ -61,24 +61,37 @@ class GoalImageView : UIView {
             object: nil,
             queue: OperationQueue.main
         ) { [weak self] _ in
-            self?.refresh()
+            DispatchQueue.main.async {
+                self?.refresh()
+            }
         }
-
         refresh()
     }
 
+    @MainActor
     private func clearGoalGraph() {
         imageView.image = UIImage(named: "GraphPlaceholder")
         currentlyShowingGraph = false
         beeLemniscateView.isHidden = true
     }
 
+    @MainActor
     private func showGraphImage(image: UIImage) {
-        imageView.image = image
-        currentlyShowingGraph = true
-        beeLemniscateView.isHidden = !(goal?.queued ?? false)
+        let duration = isThumbnail ? 0.0 : 0.4
+
+        UIView.transition(with: imageView,
+                          duration: duration,
```

Comment:
Once #614 is merged, this transition also looks great for the thumbnail as well.
```suggestion
        UIView.transition(with: imageView,
                          duration: 0.8,
```

Reply:
Given #614 is merged will let you commit this before merging.